lib/travis/sso/generic.rb

```diff
@@ -115,6 +115,7 @@ def login(request)
           user = data['user'].merge('token' =>  token)
 
           if authorized?(user)
+            template(login_page, request, provider: get_provider(user))
```

Comment:
you didn't add default get_provider, now it's not backwards compatible
+ if you don't replace provider on init - the login_page is used also in handshake , and it will fail now

Reply:
so you'd  need to update the template here( unless it's used somewhere before this point?)